src/containers/App/App.js

```diff
@@ -3,39 +3,46 @@ import { NoteArea, CreateNote } from "../";
 import { connect } from "react-redux";
 import { getNotes } from "../../thunks/";
 import PropTypes from "prop-types";
+import { showPopUp } from '../../actions';
 
 export class App extends Component {
   constructor() {
     super();
     this.state = {
-      showPopup: false
     };
   }
   
   componentDidMount = () => {
     this.props.getNotes();
   };
 
-  handleClick = () => this.setState({ showPopup: !this.state.showPopup });
+  handleClick = () => {
+    this.props.showPopUp(!this.props.shouldDisplay);
+  }
+
   render() {
-    const { showPopup } = this.state;
     return (
       <div className="App">
         <h1 className="title">Trapper-Keeper</h1>
         <div onClick={this.handleClick}>Add Note</div>
-        <CreateNote canRender={showPopup} />
+        {this.props.shouldDisplay && <CreateNote />}
         <NoteArea />
       </div>
     );
   }
 }
 
+export const mapStateToProps = state => ({
+  shouldDisplay: state.shouldDisplay
```

Comment:
So is ‘shouldDisplay’ the property in state controlled by the action ‘showPopUp’?

Reply:
Okay! I see this later on.